libs/data-access/src/lib/context/concept/action.ts

```diff
@@ -1,10 +1,11 @@
+import {SearchConceptResponseProps} from '@catalog-frontend/types';
 import {ConceptState} from './state';
 
 // add new action types here as needed: E.g. 'POPULATE' | 'UPDATE'
-export type ACTION_TYPE = 'POPULATE';
+export type ACTION_TYPE = 'POPULATE' | 'SEARCH';
 
 // add new payload types here as needed
-export type ACTION_PAYLOAD = ConceptState;
+export type ACTION_PAYLOAD = ConceptState & SearchConceptResponseProps;
```

Comment:
Her er det bare jeg som lurer: Hva betyr det når du lister de opp med `&`?

Reply:
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types
It's confusing, believe me, I know. 

Det brukes for å lage ny type fra flere eksisterende typer for å deretter kunne bruke den nye typen (her ACTION_PAYLOAD) andre steder. Tror som alternativ så kan man slette typen ACTION_PAYLOAD og istedet skrive "union type" som typen til payload. Da kommer ACTION til å se sånn ut `
export type ACTION = {type: ACTION_TYPE; payload: ConceptState | SearchConceptResponseProps};`.

Reply:
En slags multiple inheritance, ser det ut som?

Reply:
God formulering! Tja, jeg oppfatter det sånn. Først tenkte jeg det ga mer mening med "union type", men oppdaget kort tid etter det at det ikke var mulig å aksessere properties of SearchConceptResponseProps med union types. Da var koden slik: `export type ACTION_PAYLOAD = ConceptState | SearchConceptResponseProps;`

Reply:
Ja du kan sammenligne det med interface + extends men da på type. Har sett flere steder hvor det brukes type + & istedenfor interface + extends.